src/components/IstioWizards/IstioWizard.tsx

```diff
@@ -0,0 +1,265 @@
+import * as React from 'react';
+import { Button, Label, ListView, ListViewIcon, ListViewItem, Wizard } from 'patternfly-react';
+import { WorkloadOverview } from '../../types/ServiceInfo';
+import Slider from './Slider/Slider';
+import { DestinationRule, VirtualService } from '../../types/IstioObjects';
+import { serverConfig } from '../../config';
+import { authentication } from '../../utils/Authentication';
+import * as API from '../../services/Api';
+import * as MessageCenter from '../../utils/MessageCenter';
+
+type Props = {
+  show: boolean;
+  namespace: string;
+  serviceName: string;
+  workloads: WorkloadOverview[];
+  onClose: (changed: boolean) => void;
+};
+
+type WorkloadTraffic = {
+  name: string;
+  traffic: number;
+};
+
+type State = {
+  showWizard: boolean;
+  workloads: WorkloadTraffic[];
+};
+
+class IstioWizard extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      showWizard: false,
+      workloads: []
+    };
+  }
+
+  resetState = () => {
+    if (this.props.workloads.length === 0) {
+      return;
+    }
+    const wkTraffic = Math.round(100 / this.props.workloads.length);
+    const remainTraffic = 100 % this.props.workloads.length;
```

Comment:
For this iteration I didn't consider it should be more than 100 workloads versions per service.
Most scenario are <10 so I guess it could be ok for this start point.
But yeah, good point, these maths can also be checked for large scenarios.

Reply:
Added a check for this cornercase, basically in the scenario where more than 100 workloads happen, then the pre-calculation is 0 and users need to manually setup the weights.
(Instead to have this pre-calculated one).